### ISSUE TITLE
`no-unpublished-require` throws a TypeError when `"."` or `"./"` is required

### ISSUE THREAD

**Issue by fasttime** [weiran-zsd/eslint-plugin-node#48]
The rule `n/no-unpublished-require` sometimes throws a `TypeError` when `require(".")` or `require("./")` occurs in the code.

This only happens under certain circumstances:

* The `node` environment is set
* The file that contains the `require` is included in the `files` field in package.json.
* There is no file named index.js, or if package.json defines a different file in the `main` field, that file does not exist.

The smallest repro I've come up with:

**package.json**
```json
{
  "files": [
    "*.js"
  ],
  "devDependencies": {
    "eslint": "latest",
    "eslint-plugin-n": "latest"
  }
}
```

**.eslintrc**
```json
{
    "env": {
        "node": true
    },
    "plugins": [
        "n"
    ],
    "rules": {
        "n/no-unpublished-require": "error"
    }
}
```

**foo.js**
```js
require('.');
```

Running `npm install`and then `npx eslint .` in the project folder prints an error message like this:

```plain
Oops! Something went wrong! :(

ESLint: 8.22.0

TypeError: path must not be empty
Occurred while linting /home/user/my-project/foo.js:1
Rule: "n/no-unpublished-require"
```

---

**[DEMO LINK](https://stackblitz.com/edit/node-tkpu2r)**

---

I could try to fix this if desired.

**Comment by aladdin-add** [comment]
sounds it should report an eslint errror in this case - it will throw an 'module not found' error if you run your code.

**Comment by fasttime** [comment]
Yes, my code doesn't work of course. But the rule should report the linting problem instead of throwing an error.

**Comment by fasttime** [comment]
I think the root of the problem is this check where `npmignore.match` is being called with an argument that is possibly an empty string, resulting in the empty string being passed to [`ignore`](https://github.com/kaelzhang/node-ignore) that produces the `TypeError`. 

https://github.com/weiran-zsd/eslint-plugin-node/blob/d7b975a07e1b876ca2c75597d054a81564b25685/lib/util/check-publish.js#L52-L54

That's because `toRelative` returns an empty string when called with an argument that is the same as the package directory. And `require('.')` resolves to the package directory when called by a file in that location.

This could be probably fixed by making sure that `npmignore.match` is never called with an empty string. I will try to come up with a test case, and I'll create a PR if my fix works.
